MHVLibTests/Clients/MHVThingClientBlobTests.m

```diff
@@ -0,0 +1,465 @@
+//
+// MHVThingClientBlobTests.m
+// MHVLib
+//
+// Copyright (c) 2017 Microsoft Corporation. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import <XCTest/XCTest.h>
+#import "MHVCommon.h"
+#import "MHVThingClient.h"
+#import "MHVConnectionProtocol.h"
+#import "MHVMethod.h"
+#import "MHVAllergy.h"
+#import "MHVFile.h"
+#import "MHVRestRequest.h"
+#import "MHVBlobDownloadRequest.h"
+#import "MHVBlobUploadRequest.h"
+#import "MHVErrorConstants.h"
+#import "MHVServiceResponse.h"
+#import "Kiwi.h"
+
+SPEC_BEGIN(MHVThingClientBlobTests)
+
+describe(@"MHVThingClient", ^
+{
+    __block id<MHVHttpServiceOperationProtocol> requestedServiceOperationStep1;
+    __block id<MHVHttpServiceOperationProtocol> requestedServiceOperationStep2;
+    __block id<MHVHttpServiceOperationProtocol> requestedServiceOperationStep3;
+    
+    __block MHVServiceResponse *serviceResponseForStep1;
+    __block MHVServiceResponse *serviceResponseForStep2;
+    __block MHVServiceResponse *serviceResponseForStep3;
```

Comment:
Instead of naming these step1, 2, 3, can we make the names represent what they are? It's hard to tell what is going on.